### DISCUSSION TITLE
Compiling Error

### DISCUSSION THREAD

**Discussion by NotTraduce** [MrYsLab/telemetrix#82] [Q&A]
Hi, I've received this error while uploading to the Arduino UNO and I'm not very skilled in Arduino language so I'm not sure what it mean's and how to fix it, can anybody help?

Error:
 ```
C:\Users\soham\AppData\Local\Temp\cchSFvcE.ltrans0.ltrans.o: In function `twi_disable':
C:\Users\soham\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.6\libraries\Wire\src\utility/twi.c:117: undefined reference to `digitalWrite'
C:\Users\soham\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.6\libraries\Wire\src\utility/twi.c:118: undefined reference to `digitalWrite'
C:\Users\soham\AppData\Local\Temp\cchSFvcE.ltrans0.ltrans.o: In function `twi_init':
C:\Users\soham\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.6\libraries\Wire\src\utility/twi.c:88: undefined reference to `digitalWrite'
C:\Users\soham\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.6\libraries\Wire\src\utility/twi.c:89: undefined reference to `digitalWrite'
C:\Users\soham\AppData\Local\Temp\cchSFvcE.ltrans0.ltrans.o:(.rodata+0x6): undefined reference to `Print::write(unsigned char const*, unsigned int)'
c:/users/soham/appdata/local/arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino7/bin/../lib/gcc/avr/7.3.0/../../../../avr/lib/avr5/crtatmega328p.o:(.init9+0x0): undefined reference to `main'
collect2.exe: error: ld returned 1 exit status

exit status 1

Compilation error: exit status 1
```

Board: Arduino Uno

**Comment by MrYsLab** [comment]
What are you using to compile (the Arduino IDE, VS Code, or something else)?
If you use the Arduino IDE, which version and version number are you using?

Can you compile something simple, such as the blink sketch under the basics category in the examples provided with the IDE?

If you can compile Blink, can you compile the StandardFirmata sketch under the Fimata category in the examples?

**Reply by NotTraduce** [reply]
Apologies for the late reply,
I'm using the Arduino IDE, version 2.3.3-nightly20240421.

I've attempted to compile the blink sketch but it returned a compilation error.

**Reply by MrYsLab** [reply]
Have you gotten this to work with a non-nightly build?
